packages/clerk-js/src/ui/elements/FormControl.tsx

```diff
@@ -144,25 +141,19 @@ const useCalculateErrorTextHeight = (config: CalculateConfigProps = {}) => {
 
 type FormFeedbackDescriptorsKeys = 'error' | 'warning' | 'info' | 'success';
 
+type Feedback = { feedback?: string; feedbackType?: FeedbackType; shouldEnter: boolean };
+
 type FormFeedbackProps = Partial<ReturnType<typeof useFormControlFeedback>['debounced'] & { id: FieldId }> & {
   elementDescriptors?: Partial<Record<FormFeedbackDescriptorsKeys, ElementDescriptor>>;
 };
 
-const delay = 350;
-
 export const FormFeedback = (props: FormFeedbackProps) => {
-  const { id, elementDescriptors } = props;
-  const errorMessage = useFieldMessageVisibility(props.errorText, delay);
-  const successMessage = useFieldMessageVisibility(props.successfulText, delay);
-  const informationMessage = useFieldMessageVisibility(props.informationText, delay);
-  const warningMessage = useFieldMessageVisibility(props.warningText, delay);
+  const { id, elementDescriptors, feedback, feedbackType = 'info' } = props;
+  const feedbacksRef = useRef<{
+    a?: Feedback;
+    b?: Feedback;
+  }>({ a: { feedback, feedbackType, shouldEnter: true }, b: undefined });
```

Comment:
I need two different messages in the state here. These two take turns representing the most current message, while the other one fades out. This is just simplifying what we had before, as we had 4 components (one for each feedback type). These would fade in and out as the error changed. We don't need 4, just 2, since the error just swaps out. Tbh I couldn't think of a name here.

Reply:
Makes sense! `a` and `b` seem reasonable in this case then if they don't have any meaningful distinction.